pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,396 @@
+  
+  Same Syntax for decorators [`@post`](http://apidocs.loopback.io/@loopback%2fcore/#802)
+  , [`@put`](http://apidocs.loopback.io/@loopback%2fcore/#806)
+  , [`@patch`](http://apidocs.loopback.io/@loopback%2fcore/#810)
+  , [`@del`](http://apidocs.loopback.io/@loopback%2fcore/#814)
+
+  You can call these suger operation decorators as a shortcut of `@operation`, for example:
+  ```ts
+  class MyController {
+    @get('/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+
+  is equivalent to
+
+  ```ts
+  class MyController {
+    @operation('GET', '/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+  
+  For more usage, refer to [Routing to Controllers](controllers.htm#routing-to-controllers)
+
+### Parameter Decorator
+
+  Syntax: [`@param(paramSpec: ParameterObject)`]()
+
+  Optional Pattern: `@param.${in}.${type}(${name})`
+
+  `@param` can be applied to method itself or specific parameters, it describes
+  an input parameter of a Controller method.
+
+  For example:
+
+  ```ts
+    // example 1: decorator `@param ` applied on method level
+    class MyController {
+      @get('/')
+      @param(offsetSpec)
+      @param(pageSizeSpec)
+      list(offset?: number, pageSize?: number) {}
+    }
+  ```
+
+  or
+  
+  ```ts
+    // example 2: decorator `@param` applied on parameter level
+    class MyController {
+      @get('/')
+      list(
+        @param(offsetSpec) offset?: number,
+        @param(pageSizeSpec) pageSize?: number,
+      ) {}
+    }
+  ```
+
+  By the first example you apply multiple decorators to a single declaration, 
+  their sequence is important, a `@param` decorator must be applied after 
+  an operation decorator. Know more about TypeScript decorator composition, 
+  refer to [TypeScript Decorator Documentation](https://www.typescriptlang.org/docs/handbook/decorators.html)
+
+  Please note method level `@oaram` and parameter level `@param` are mutually exclusive,
+  you can not mix and apply them to the same parameter.
+  
+  You can also use this pattern to make it quicker to define params: `@param.${in}.${type}(${name})`
+  
+  - in: one of the following values: `query`, `header`, `path`, `formData`, `body`
+  - type: one of the following values:  `string`, `number`, `boolean`, `integer`
+  - name: a `string`, name of the parameter
+
+  So an example would be `@param.query.number('offset')`. 
+  You can find the specific usage in [Writing Controller methods](controller.htm#writing-controller-methods)
+  
+## Dependency Injection
+
+`@inject` is a decorator to annotate method arguments for automatic injection by 
+LoopBack IoC container. 
+
+The injected values are applied to a constructed instance, so it can only be used on 
+non-static properties or constructor parameters of a Class.
+
+The `@inject` decorator allows you inject dependencies bound to the application context, 
+to the per-request contextor, to any implementation of the [Context](#context) object. 
```

Comment:
`The @inject decorator allows you to inject dependencies bound to any Context, such as an Application instance or a request context instance.`

I can't add the backticks in the example for `@inject` so keep that in mind. :P